src/compiler/checker.ts

```diff
@@ -4504,8 +4542,8 @@ namespace ts {
                         parameters.push(paramSymbol);
                     }
 
-                    if (param.type && param.type.kind === SyntaxKind.StringLiteralType) {
-                        hasStringLiterals = true;
+                    if (param.type && param.type.kind === SyntaxKind.LiteralType) {
+                        hasLiteralTypes = true;

```

Comment:
I *don't* think you want to do this. This adds broken behavior to all literal types (due to floating up specialized overloads) instead of limiting it to string literals.